drivers/soundwire/stream.c

```diff
@@ -1882,9 +1882,16 @@ int sdw_stream_add_master(struct sdw_bus *bus,
 			  unsigned int num_ports,
 			  struct sdw_stream_runtime *stream)
 {
+	struct sdw_slave_prop *slave_prop;
 	struct sdw_master_runtime *m_rt;
+	struct sdw_slave_runtime *s_rt;
+	struct sdw_port_runtime *p_rt;
 	bool alloc_master_rt = false;
+	bool is_lane_set = false;
+	int lane[SDW_MAX_LANES];
+	int port_index = 0;
 	int ret;
+	int i;
```

Comment:
"A controller should use the lane that connects to the peripheral"

That wording is controversial. 

First we are talking about a Manager. Controllers don't have a concept of Lanes, they are made of one or more Managers. Each Manager can have one or more Lanes. Different level.

Also, you could perfectly well have multiple lanes connecting a Manager and a Peripheral. For example to transmit 8ch 192kHz 24 bits we would need 4 lanes, one for each stereo pair.

Reply:
The commit is to map the lane that connects to a specific peripheral lane. For example, peripheral lane 1 connects to manager lane 2. We already know which peripheral lane is used and need to get the manager lane by `slave_prop->lane_maps`.